lib/gui/components/modal/services/modal.js

```diff
@@ -55,23 +59,33 @@ module.exports = function($uibModal, $q) {
     return {
       close: modal.close,
       result: $q((resolve, reject) => {
-        modal.result
-          .then(resolve)
-          .catch((error) => {
+        modal.result.then((value) => {
+          AnalyticsService.logEvent('Modal accepted', value);
+
+          resolve(value);
+        }).catch((error) => {
+
+          // For some annoying reason, UI Bootstrap Modal rejects
+          // the result reason if the user clicks on the backdrop
+          // (e.g: the area surrounding the modal).
+          if (error === 'escape key press' || error === 'backdrop click') {
+            AnalyticsService.logEvent('Modal rejected', {
+              method: error
+            });
 
             // Bootstrap doesn't 'resolve' these but cancels the dialog;
-            // therefore call 'resolve' here applied to 'false'.
-            if (error === 'escape key press' || error === 'backdrop click') {
-              resolve();
+            // therefore call 'resolve' here.
+            resolve();
+
+          } else {
+            AnalyticsService.logEvent('Modal rejected', {
+              value: error
```

Comment:
I suppose the best way to answer that question is to ask what kind of things we want to analyse after all the statistics have been gathered? (and therefore whether we're gathering the "right" statistics).

But that's something I have no idea about.